src/ast.js

```diff
@@ -7,6 +7,9 @@ const regenerate = require('regenerate');
 class Alternation {
   constructor(a, b) {
     this.precedence = 1;
+    if (b.toString().length > a.toString().length) {
+      [a, b] = [b, a];
+    }
```

Comment:
Not sure this will always work. What if `a` or `b` were itself another alternation, for example, in the case of 3 or more options. Then the stringified version of that alternation might be longer than another single option, but perhaps longer than the individual options in the existing alternation. For example:

```
a = 'a|b|c|d'
b = 'ab'
```

In that case, b should go first but the stringified a is longer.